src/distribution_metadata.c

```diff
@@ -411,10 +452,16 @@ IsDistributedTable(Oid tableId)
 
 	SPI_connect();
 
-	spiStatus = SPI_execute_with_args("SELECT NULL "
-									  "FROM pgs_distribution_metadata.partition "
-									  "WHERE relation_id = $1", argCount, argTypes,
-									  argValues, NULL, false, 1);
+	if (spiPlan == NULL)
+	{
+		spiPlan = SPI_prepare("SELECT NULL FROM pgs_distribution_metadata.partition "
+							  "WHERE relation_id = $1", argCount, argTypes);
+
+		spiStatus = SPI_keepplan(spiPlan);
+		Assert(spiStatus == 0);
+	}
+
+	spiStatus = SPI_execute_plan(spiPlan, argValues, NULL, false, 1);
```

Comment:
Given that it wasn't before, I don't want to change it now. I could have had a reason for it (cargo-culting!), but I don't remember any more. Might have had something to do with undesirable snapshot semantics, but I'm honestly speculating at this point.

Were there any blockers for :shipit:?